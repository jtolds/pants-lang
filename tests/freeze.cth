outside_func := { |x| function {
  outside_return := return
  inside_func := { |y| function {
    outside_return y
  }}
  inside_func 3
  2
}}

print outside_func(1)

print.

outside_func := { |x| function {
  outside_return := freeze(return)
  inside_func := { |y| function {
    outside_return y
  }}
  inside_func 3
  2
}}

print outside_func(1)
