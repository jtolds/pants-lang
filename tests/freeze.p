outside_func = { |x| function {
  outside_return = return
  inside_func = { |y| function {
    outside_return y
  }}
  inside_func 3
  2
}}

println outside_func(1)

println.

outside_func = { |x| function {
  outside_return = freeze(return)
  inside_func = { |y| function {
    outside_return y
  }}
  inside_func 3
  2
}}

println outside_func(1)
