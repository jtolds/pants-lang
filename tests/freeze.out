2

3

return code: 0
